apix-impl/src/main/java/eu/xenit/apix/alfresco/search/SearchFacetsServiceImpl.java

```diff
@@ -332,4 +351,26 @@ public static FacetLabelDisplayHandler createFromProperty(PropertyDefinition pro
         }
 
     }
+
+    public static class CategoryFacetLabelDisplayHandler implements FacetLabelDisplayHandler {
+
+        private NodeService nodeService;
+
+        private CategoryFacetLabelDisplayHandler(NodeService nodeService) {
+            this.nodeService = nodeService;
+        }
+
+        @Override
+        public FacetLabel getDisplayLabel(String value) {
+            NodeRef nodeRef = new NodeRef(value);
+            try {
+                Serializable nameProperty = nodeService.getProperty(nodeRef, ContentModel.PROP_NAME);
```

Comment:
I guess I can do that although I am not sure if the name property can ever be null. if you create a node in the javascript console and choose null as the name of the node it will just use the guid of the noderef as the node name.